Framework/Algorithms/src/GetDetectorOffsets.cpp

```diff
@@ -204,16 +212,33 @@ double GetDetectorOffsets::fitSpectra(const int64_t s, bool isAbsolbute) {
 
   // std::vector<double> params = fit_alg->getProperty("Parameters");
   API::IFunction_sptr function = fit_alg->getProperty("Function");
+
   double offset = function->getParameter(3); // params[3]; // f1.PeakCentre
-  offset = -1. * offset * m_step / (m_dreference + offset * m_step);
-  // factor := factor * (1+offset) for d-spacemap conversion so factor cannot be
-  // negative
 
-  if (isAbsolbute) {
+  if (mode == offset_mode::signed_offset) {
```

Comment:
Can you add a test for this new mode https://github.com/mantidproject/mantid/blob/master/Framework/Algorithms/test/GetDetectorOffsetsTest.h